cypress/integration/discounts/vouchers/createVouchers.js

```diff
@@ -0,0 +1,351 @@
+    });
+
+    it("should be able to create free shipping voucher. TC: SALEOR_1903", () => {
+      const voucherCode = `${startsWith}${faker.datatype.number()}`;
+      const expectedAmount = productPrice;
+      let checkout;
+
+      loginAndCreateCheckoutForVoucherWithDiscount({
+        discount: discountOptions.SHIPPING,
+        voucherCode,
+        channelName: defaultChannel.name,
+        dataForCheckout
+      })
+        .then(({ addPromoCodeResp, checkout: checkoutResp }) => {
+          expect(addPromoCodeResp.checkout.totalPrice.gross.amount).to.be.eq(
+            expectedAmount
+          );
+          dataForCheckout.voucherCode = voucherCode;
+          checkout = checkoutResp;
+          addPayment(checkout.id);
+        })
+        .then(() => {
+          completeCheckout(checkout.id);
+        })
+        .then(({ order }) => {
+          expect(order.id).to.be.ok;
+        });
+    });
+
+    it("should be able to create voucher not available for selected channel. TC: SALEOR_1904", () => {
+      const randomName = `${startsWith}${faker.datatype.number()}`;
+      const voucherValue = 50;
+
+      cy.clearSessionData()
+        .loginUserViaRequest()
+        .visit(urlList.vouchers);
+      cy.softExpectSkeletonIsVisible();
+      createVoucher({
+        voucherCode: randomName,
+        voucherValue,
+        discountOption: discountOptions.PERCENTAGE,
+        channelName: createdChannel.name
+      });
+      dataForCheckout.voucherCode = randomName;
+      createCheckoutWithVoucher(dataForCheckout).then(
+        ({ addPromoCodeResp }) => {
+          const errorField = addPromoCodeResp.checkoutErrors[0].field;
+          expect(errorField).to.be.eq("promoCode");
+        }
+      );
+    });
+
+    it("should be able to create voucher with limited number of times discount can be used in total. TC: SALEOR_1907", () => {
+      const voucherCode = `${startsWith}${faker.datatype.number()}`;
+      const voucherValue = 50;
+      const usageLimit = 1;
+      let firstCheckout;
+
+      loginAndCreateCheckoutForVoucherWithDiscount({
+        discount: discountOptions.PERCENTAGE,
+        voucherValue,
+        voucherCode,
+        channelName: defaultChannel.name,
+        dataForCheckout,
+        usageLimit
+      })
+        .then(({ checkout }) => {
+          firstCheckout = checkout;
+          dataForCheckout.voucherCode = voucherCode;
+          addPayment(firstCheckout.id);
+        })
+        .then(() => {
+          completeCheckout(firstCheckout.id);
+        })
+        .then(() => {
+          createCheckoutWithVoucher(dataForCheckout);
+        })
+        .then(({ addPromoCodeResp }) => {
+          const errorField = addPromoCodeResp.checkoutErrors[0].field;
+          expect(errorField, "error in promo code should occur").to.be.eq(
+            "promoCode"
+          );
+        });
+    });
+
+    it("should be able to create voucher with limit to one use per customer. TC: SALEOR_1908", () => {
```

Comment:
Why we do not check in first checkout if promoCode is added and there is no error and order is created with discount and why there are 3 createCheckout?

Reply:
3rd checkout is to check if other (not logged in) customer still can use that voucher in checkout. I added comment in this TC. I added check if there is no errors in first checkout.